diff -Nru mysql/storage/innobase/include/fts0tokenize.h mysql.patched/storage/innobase/include/fts0tokenize.h
--- mysql/storage/innobase/include/fts0tokenize.h	2019-08-14 16:25:47.000000000 +0200
+++ mysql.patched/storage/innobase/include/fts0tokenize.h	2019-08-26 10:06:55.157492259 +0200
@@ -39,7 +39,7 @@
 inline bool true_word_char(char c, bool extra_word_chars, char ch) {
   bool result =
       ((extra_word_chars) ? !((c) & (_MY_SPC))
-                          : ((c) & (_MY_U | _MY_L | _MY_NMR) || (ch) == '_'));
+                          : ((c) & (_MY_U | _MY_L | _MY_NMR) || (ch) == '_' || (ch) == '.' || (ch) == ','));
   return result;
 }
 
diff -Nru mysql/storage/myisam/ftdefs.h mysql.patched/storage/myisam/ftdefs.h
--- mysql/storage/myisam/ftdefs.h	2019-08-14 16:25:47.000000000 +0200
+++ mysql.patched/storage/myisam/ftdefs.h	2019-08-26 10:07:41.677288771 +0200
@@ -36,7 +36,7 @@
 #include "storage/myisam/queues.h"
 
 #define true_word_char(ctype, character) \
-  ((ctype) & (_MY_U | _MY_L | _MY_NMR) || (character) == '_')
+  ((ctype) & (_MY_U | _MY_L | _MY_NMR) || (character) == '_' || (character) == '.' || (character) == ',')
 #define misc_word_char(X) 0
 
 #define FT_MAX_WORD_LEN_FOR_SORT 31
