diff -Nru mysql/storage/innobase/handler/i_s.cc mysql.patched/storage/innobase/handler/i_s.cc
--- mysql/storage/innobase/handler/i_s.cc	2019-08-14 16:25:47.000000000 +0200
+++ mysql.patched/storage/innobase/handler/i_s.cc	2019-08-20 21:40:57.766293057 +0200
@@ -7536,12 +7536,12 @@
     STRUCT_FLD(descr, "InnoDB CHANGED_PAGES table"),
     STRUCT_FLD(license, PLUGIN_LICENSE_GPL),
     STRUCT_FLD(init, i_s_innodb_changed_pages_init),
+    NULL,
     STRUCT_FLD(deinit, i_s_common_deinit),
-    nullptr,
-    STRUCT_FLD(version, 0x0100 /* 1.0 */),
-    STRUCT_FLD(status_vars, nullptr),
-    STRUCT_FLD(system_vars, nullptr),
-    STRUCT_FLD(__reserved1, nullptr),
+    STRUCT_FLD(version, 0x0100),
+    STRUCT_FLD(status_vars, NULL),
+    STRUCT_FLD(system_vars, NULL),
+    STRUCT_FLD(__reserved1, NULL),
     STRUCT_FLD(flags, 0UL),
 };
 
