diff -Nru percona-server-5.7.22-22/plugin/keyring/common/system_keys_container.cc percona-server-5.7.22-22.patched/plugin/keyring/common/system_keys_container.cc
--- percona-server-5.7.22-22/plugin/keyring/common/system_keys_container.cc	2018-05-29 23:11:24.000000000 +0200
+++ percona-server-5.7.22-22.patched/plugin/keyring/common/system_keys_container.cc	2018-06-20 12:37:20.839118808 +0200
@@ -125,7 +125,7 @@
   if (is_system_key_with_version(key, system_key_id, key_version))
   {
     if (system_key_id_to_system_key.count(system_key_id) == 0) // add a new system key
-      system_key_id_to_system_key.insert(std::make_pair<std::string, System_key_adapter*>(system_key_id, new System_key_adapter(key_version, key)));
+      system_key_id_to_system_key.insert(std::make_pair(system_key_id, new System_key_adapter(key_version, key)));
     else
       update_system_key(key, system_key_id, key_version);
   }
