diff -Nru mysql/storage/innobase/row/row0mysql.cc mysql.patched/storage/innobase/row/row0mysql.cc
--- mysql/storage/innobase/row/row0mysql.cc	2018-05-29 23:11:24.000000000 +0200
+++ mysql.patched/storage/innobase/row/row0mysql.cc	2018-09-11 10:38:56.300286826 +0200
@@ -5888,6 +5888,17 @@
 		goto funct_exit;
 	}
 
+	/* Wait for background fts sync to finish */
+	for (retry = 1; dict_fts_index_syncing(table); ++retry) {
+		DICT_BG_YIELD(trx);
+		if (retry % 100 == 0) {
+			ib::info() << "Unable to rename table " << old_name
+				<< " to new name " << new_name
+				<< " because FTS sync is running on table."
+				<< " Retrying";
+		}
+	}
+
 	/* We use the private SQL parser of Innobase to generate the query
 	graphs needed in updating the dictionary data from system tables. */
 
