diff -u -r centerim-4.22.10/configure.ac centerim-4.22.10.new/configure.ac
--- centerim-4.22.10/configure.ac	2010-11-23 20:03:04.000000000 +0100
+++ centerim-4.22.10.new/configure.ac	2017-11-23 17:12:25.544940869 +0100
@@ -384,10 +384,11 @@
 AC_CHECK_LIB(socket, socket)
 AC_CHECK_LIB(nsl, gethostbyname)
 
+PKG_PROG_PKG_CONFIG
 AX_WITH_CURSES
-if test "x$ax_cv_ncurses" = "xyes"
+if test "x$ax_cv_ncurses" = "xyes" -o "x$ax_cv_ncursesw" = "xyes"
 then
-                LIBS="$LIBS $CURSES_LIB"
+                LIBS="$LIBS $CURSES_LIBS"
 else
                 AC_MSG_ERROR([You need ncurses/ncursesw library and header files.])
 fi
