From 556a2baf1c3f9e7620f99441730bae09b0e175da Mon Sep 17 00:00:00 2001
From: madaidan <50278627+madaidan@users.noreply.github.com>
Date: Fri, 22 May 2020 01:13:06 +0000
Subject: [PATCH] Mention sysctl

---
 security/Kconfig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/security/Kconfig b/security/Kconfig
index 8ad042218d62..040a35869eb3 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -62,6 +62,11 @@ config SECURITY_HARDEN_IPC
 	  who found hundreds of such insecure usages. Processes with
 	  CAP_IPC_OWNER are still permitted to access these IPC objects.
 
+	  This setting can be overridden at runtime via the kernel.harden_ipc
+	  sysctl.
+
+	  If unsure, say Y.
+
 config SECURITY
 	bool "Enable different security models"
 	depends on SYSFS
