diff -Nru zabbix-5.0.0rc1/configure zabbix-5.0.0rc1.patch/configure
--- zabbix-5.0.0rc1/configure	2020-05-05 15:29:56.000000000 +0200
+++ zabbix-5.0.0rc1.patch/configure	2020-05-06 10:18:46.534941614 +0200
@@ -14161,7 +14161,7 @@
 CURL_SSL_CERT_LOCATION="${datadir}/zabbix/ssl/certs"
 CURL_SSL_KEY_LOCATION="${datadir}/zabbix/ssl/keys"
 
-LOAD_MODULE_PATH="${libdir}/modules"
+LOAD_MODULE_PATH="${libdir}/zabbix"
 
 
 
